python/pyspark/sql/connect/dataframe.py

```diff
@@ -1623,6 +1624,10 @@ def __getitem__(self, item: Union[int, str, Column, List, Tuple]) -> Union[Colum
             alias = self._get_alias()
             if self._plan is None:
                 raise SparkConnectException("Cannot analyze on empty plan.")
+
+            if "." not in item and "*" not in item and "`" not in item and item not in self.columns:
```

Comment:
similar to https://github.com/apache/spark/commit/91e97f92fe76f9718cd16af0c761d5530bdb37ee, but need to cases like `*`, `a.*`, ``aa.bb``

Reply:
Should we maybe just:

```python
col = _to_col_with_plan_id(
    col=alias if alias is not None else item,
    plan_id=self._plan._plan_id,
)
self.select(col)
return col
```

Reply:
good idea!